x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_management/components/analytics_list/action_delete.tsx

```diff
@@ -109,9 +191,40 @@ export const DeleteAction: FC<DeleteActionProps> = ({ item }) => {
           >
             <p>
               {i18n.translate('xpack.ml.dataframe.analyticsList.deleteModalBody', {
-                defaultMessage: `Are you sure you want to delete this analytics job? The analytics job's destination index and optional Kibana index pattern will not be deleted.`,
+                defaultMessage: `Are you sure you want to delete this analytics job?`,
               })}
             </p>
+            <EuiFlexGroup direction="column" gutterSize="none">
+              <EuiFlexItem>
+                {userCanDeleteIndex && (
+                  <EuiSwitch
+                    style={{ paddingBottom: 10 }}
+                    label={i18n.translate(
+                      'xpack.ml.dataframe.analyticsList.deleteTargetIndexTitle',
+                      {
+                        defaultMessage: 'Delete destination index',
+                      }
+                    )}
+                    checked={deleteTargetIndex}
+                    onChange={e => toggleDeleteTargetIndex(!deleteTargetIndex)}
+                  />
+                )}
+              </EuiFlexItem>
+              <EuiFlexItem>
+                {userCanDeleteIndex && indexPatternExists && (
+                  <EuiSwitch
+                    label={i18n.translate(
+                      'xpack.ml.dataframe.analyticsList.deleteTargetIndexTitle',
+                      {
+                        defaultMessage: 'Delete index pattern',
```

Comment:
I think it is worth clarifying which index pattern is deleted here, using a message of `Delete destination index pattern {indexName}`

Reply:
Done in https://github.com/elastic/kibana/pull/66934/commits/e5e618f4d8665970bea50f11f12b63be43e5bd2f